src/vdaf/prio3.rs

```diff
@@ -223,6 +223,165 @@ where
     pub fn verifier_len(&self) -> usize {
         self.typ.verifier_len()
     }
+
+    fn setup_with_rand_source<R: Fn(&mut [u8]) -> Result<(), getrandom::Error>>(
```

Comment:
nit: this trait bound occurs in a few places, and its signature doesn't make its purpose (which is to fill a buffer with random values, right?) obvious. A well-named type alias could make this more clear.

And if you did that, then you could possibly define the type alias to also require `Copy`, which you end up requiring everywhere anyway, I assume because you have to pass the `rand_source` function by value into various places.

Reply:
I'm not sure this works the way you expect. Maybe I'm missing something?

```
$ cargo test
   Compiling prio v0.6.1 (/Users/chris/github.com/abetterinternet/libprio-rs)
error[E0404]: expected trait, found type alias `RandSource`
  --> src/vdaf/prg.rs:31:39
   |
31 |     pub(crate) fn from_rand_source<R: RandSource>(
   |                                       ^^^^^^^^^^ type aliases cannot be used as traits

warning: trait objects without an explicit `dyn` are deprecated
  --> src/vdaf/prg.rs:19:30
   |
19 | pub(crate) type RandSource = Fn(&mut [u8]) -> Result<(), getrandom::Error>;
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `dyn`: `dyn Fn(&mut [u8]) -> Result<(), getrandom::Error>`
```

Reply:
I was hoping this would Just Work. I think maybe with some trait definitions it could be made to work, but it's really not worth the trouble. Let's just move ahead with the `R: Fn(&mut [u8]) -> Result<(), getrandom::Error>` bound.

Reply:
I think you could get rid of the type variable R, create a type alias for the function pointer type `fn(&mut [u8]) -> Result<(), getrandom::Error>`, and have the rand_source arguments be of that type. (you'll need to coerce `getrandom::getrandom as MyTypeAlias` to get a function pointer)